queries/kcl/injections.scm

```diff
@@ -0,0 +1,33 @@
+(call_expr
+  function: (selector_expr
+    (identifier) @_regex)
+  arguments: (argument_list
+    (_)
+    .
+    (string
+      (string_content) @injection.content))
+  (#eq? @_regex "regex")
+  (#set! injection.language "regex"))
+
+(call_expr
+  function: (selector_expr
+    (identifier) @_regex
+    (select_suffix
+      (identifier) @_fn (#eq? @_fn "compile")))
+  arguments: (argument_list
+    .
+    (string
+      (string_content) @injection.content))
+  (#eq? @_regex "regex")
+  (#set! injection.language "regex"))
+
+((comment) @injection.content
+  (#set! injection.language "comment"))
+
+; Unfortunately, rst doesn't work with the indented docstrings, markdown_inline at least provides some highlighting
```

Comment:
Sorry, but that's a bad idea. All manner of things use injection queries, and rely on the language being correct.

Reply:
Ok, I removed it. It's a shame that #2212 is not solved, all auto-generated docstrings (from imported xrd's etc.) are in Restructured Text but injection does not work because of the indentation. Markdown_inline works actually quite well due to the similarities between rst and markdown although it would only highlight part of it (italic, bold, links, ... but not headings)

Reply:
If it's rst, it should be injected as rst and nothing else. I don't quite see the relevance of the linked issue?

Reply:
The doc-comment is indented, so each line has a whitespace prefix. Indented rst implies a quote I think, at least it is not or almost not highlighted because of the indentation. I thought that #2212 refers to a similar issue with markdown. But anyway, you are probably right that it's better to not inject anything than injecting something else that just partly works.